ember.md

```diff
@@ -0,0 +1,461 @@
+# Ember.js Style Guide
+
+## Table Of Contents
+
+* [General](#general)
+* [Organizing your modules](#organizing-your-modules)
+* [Models](#models)
+* [Controllers](#controllers)
+* [Templates](#templates)
+* [Routing](#routing)
+* [Ember data](#ember-data)
+
+## General
+
+### Import what you use, do not use globals
+
+For Ember Data, we should import `DS` from `ember-data`, and then
+destructure our desired modules.
+For Ember, use destructuring [as Ember's internal organization is
+not intuitive and difficult to grok, and we should wait until Ember has been
+correctly modularized.](https://github.com/ember-cli/ember-cli-shims/issues/53)
+
+Once Ember has officially adopted shims, we will prefer shims over
+destructuring.
+
+```javascript
+// Good
+
+import Ember from 'ember';
+import DS from 'ember-data';
+
+const {
+  computed,
+} = Ember;
+
+const {
+  attr,
+} = DS;
+
+export default DS.Model.extend({
+  firstName: attr('string'),
+  lastName: attr('string'),
+
+  fullName: computed('firstName', 'lastName', function() {
+    // Code
+  }),
+});
+
+// Bad
+
+export default DS.Model.extend({
+  firstName: DS.attr('string'),
+  lastName: DS.attr('string'),
+
+  fullName: Ember.computed('firstName', 'lastName', function _fullName() {
+    // Code
+  }),
+});
+```
+
+### Don't use Ember's prototype extensions
+
+Avoid Ember's `Date`, `Function` and `String` prototype extensions. Prefer the
+corresponding functions from the `Ember` object.
+
+```javascript
+// Good
+
+const {
+  String: { w },
+  computed,
+  on,
+} = Ember;
+
```

Comment:
I think we should move `// Good` here, between `const` and `export` because the `const` declarations are needed for both good and bad examples.  Same with other sections.